### ISSUE TITLE
Generated premake projects expects TM_SDK_DIR to have a layout that doesnt match the engine source tree

### ISSUE THREAD

**Issue by photex** [OurMachinery/themachinery-public#621]
I realize that this isn't perhaps a generally applicable or useful issue, but it is something I ran into as a new user to both premake and the-machinery.

I built the-machinery from source in order to apply a fix to font search paths while waiting for the next release. After generating a plugin I was unable to build it since I set TM_SDK_DIR to the path to the cloned repo. The premake script generated needed some adjustments to include paths and library paths.

I also attempted to package the machinery but this wasn't possible due to the unit tests failing.

To summarize, in order to get things working I just needed to fiddle with the premake script. It could be helpful to find some way to not require this (zig build perhaps?) :D

**Comment by photex** [comment]
Yep, the plugins in the main repo all built fine. But the premake script created by the editor (File -> New Plugin) needed the adjustment.

**Comment by simon-ourmachinery** [comment]
yes that is because it is targeted to none source code access customers. I understand the issue and will fix it at some point

**Comment by jamesmintram** [comment]
I've customized my premake file here: https://github.com/jamesmintram/om_sample_test/blob/master/plugins/premake5.lua to handle both cases.

Could serve as inspiration/a template for people to use.

**Comment by simon-ourmachinery** [comment]
will be addressed in the next release see https://github.com/OurMachinery/themachinery/commit/7bf235ef97dd9742af198e56eb5860c71a214719
